lib/block-supports/index.php

```diff
@@ -96,7 +96,7 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 		$block_root->setAttribute( 'style', esc_attr( implode( '; ', $new_styles ) . ';' ) );
 	}
 
-	return $dom->saveHtml( $block_root );
+	return trim( str_replace( array( $wrapper_left, $wrapper_right ), '', $dom->saveHtml() ) );
```

Comment:
It seems somewhat unsafe to rely on the original HTML input to be identical after DOM serialization. I'd expect there to be some changes being made like addition of whitespace or changing attribute order or something at some point.

Reply:
You might want to consider what we are doing in the AMP plugin:

https://github.com/ampproject/amp-wp/blob/fc0373defb7cc8e5c4331d57d9140e8bd1454ac3/includes/utils/class-amp-dom-utils.php#L178-L193

Reply:
That's a good point, thanks for bringing it up. I've implemented an alternative that concatenates the child nodes of the body element that I believe improves on the original implementation from https://github.com/WordPress/gutenberg/pull/24645.